routing/pathfind_test.go

```diff
@@ -119,6 +126,209 @@ func makeTestGraph() (*channeldb.ChannelGraph, func(), error) {
 	return cdb.ChannelGraph(), cleanUp, nil
 }
 
+// parseDescribeGraph parses a JSON file created by using 'lncli describegraph',
+// and populates a ChannelGraph with this graph info.
+func parseDescribeGraph(path string, sourceNodeAlias string) (
```

Comment:
It was mentioned to me that this parsing could be done by doing an `unmarshal` of the JSON into a `lnrpc.ChannelGraph`, the reverse operation of the marshalling that is done when the JSON is created: https://github.com/lightningnetwork/lnd/blob/7abdd30a878efb3673ce4dfe0070570a6c9444c8/rpcserver.go#L2387

Reply:
Can you elaborate on this? Do you mean we should write an [unmarshal interface](https://golang.org/pkg/encoding/json/#Unmarshal)? I haven't used it before so I can't see any benefits of doing that?

Also, sorry that your commit couldn't be cherry picked, for some reason git refused to allow me to rebase your commit(invalid author or something).

By the way, I have made all the other changes, so feel free to review again.

Reply:
After some more looking up, I believe it's better to use an external script to encode the describe graph output into one with format like the current internal tests. Besides being smaller in size, it's much more consistent and would avoid code duplicity. I doubt it's going to be changed very often as well. Let me know what you think.